core/src/com/unciv/logic/map/tile/TileImprovementFunctions.kt

```diff
@@ -209,11 +211,20 @@ class TileImprovementFunctions(val tile: Tile) {
             else -> {
                 tile.improvementIsPillaged = false
                 tile.improvement = improvementName
+                improvementFieldHasChanged = true
 
                 removeCreatesOneImprovementMarker()
             }
         }
 
+        if (improvementFieldHasChanged) {
+            for (civ in tile.tileMap.gameInfo.civilizations) {
+                if (civ.isDefeated() || !civ.isMajorCiv() || civ.isSpectator()) continue
```

Comment:
I'm pretty sure spectators are not major civs
```suggestion
                if (civ.isDefeated() || !civ.isMajorCiv()) continue
```

Reply:
You can just "commit suggestion" :)